Last-Update: 2013-01-07
Author: Dmitry Smirnov <onlyjob@member.fsf.org>
Forwarded: no
Description: removes bundled "libltdl-dev" in order to use system one.

--- a/FreeOrion/GG/CMakeLists.txt
+++ b/FreeOrion/GG/CMakeLists.txt
@@ -230,41 +230,8 @@
     message(FATAL_ERROR "Freetype library not found.")
 endif ()
 
 
-########################################
-# Header Generation                    #
-########################################
-if (NOT EXISTS ${CMAKE_HOME_DIRECTORY}/libltdl/config.h)
-    if (UNIX)
-        message("-- Configuring libltdl using \"libltdl/configure\" ...")
-        execute_process(
-            COMMAND ${CMAKE_HOME_DIRECTORY}/libltdl/configure
-            WORKING_DIRECTORY ${CMAKE_HOME_DIRECTORY}/libltdl
-            OUTPUT_QUIET
-        )
-    else ()
-        file(WRITE ${CMAKE_HOME_DIRECTORY}/libltdl/config.h
-             "/* WARNING: Generated by GG's build system.  All local changes will be lost! */
-#define error_t int
-#define HAVE_STDIO_H 1
-#define HAVE_STDLIB_H 1
-#define HAVE_STRING_H 1
-#define HAVE_CTYPE_H 1
-#define HAVE_MEMORY_H 1
-#define HAVE_ERRNO_H 1
-#define __WIN32__
-#define HAVE_MEMCPY 1
-#define HAVE_MEMMOVE 1
-#define LTDL_OBJDIR \".libs\"
-#define LTDL_DLOPEN_DEPLIBS 1
-#define LTDL_SHLIBPATH_VAR \"PATH\"
-#define LTDL_SHLIB_EXT \".dll\"
-"
-        )
-    endif ()
-endif ()
-
 configure_file(
     ${CMAKE_HOME_DIRECTORY}/cmake/Config.h.in
     ${CMAKE_HOME_DIRECTORY}/GG/Config.h
     @ONLY
--- a/FreeOrion/GG/src/CMakeLists.txt
+++ b/FreeOrion/GG/src/CMakeLists.txt
@@ -49,10 +49,8 @@
     dialogs/ColorDlg.cpp
     dialogs/FileDlg.cpp
     dialogs/ThreeButtonDlg.cpp
 
-    ../libltdl/ltdl.c
-
     adobe/adam.cpp
     adobe/adam_evaluate.cpp
     adobe/adam_parser.cpp
     adobe/any_regular.cpp
@@ -161,15 +159,8 @@
         adobe/future/widgets/sources/platform_window.cpp
     )
 endif ()
 
-set_source_files_properties(${CMAKE_HOME_DIRECTORY}/libltdl/ltdl.c PROPERTIES COMPILE_DEFINITIONS HAVE_CONFIG_H=1)
-execute_process(
-    COMMAND ${CMAKE_COMMAND} -E copy_if_different ${CMAKE_HOME_DIRECTORY}/libltdl/ltdl.h ${CMAKE_HOME_DIRECTORY}/GG/ltdl.h
-    COMMAND ${CMAKE_COMMAND} -E copy_if_different ${CMAKE_HOME_DIRECTORY}/libltdl/config.h ${CMAKE_HOME_DIRECTORY}/GG/ltdl_config.h
-)
-set_source_files_properties(${CMAKE_HOME_DIRECTORY}/GG/ltdl.h ${CMAKE_HOME_DIRECTORY}/GG/ltdl_config.h PROPERTIES GENERATED true)
-
 set(THIS_LIB_LINK_LIBS ${Boost_LIBRARIES} ${OPENGL_LIBRARIES} ${FREETYPE_LIBRARIES})
 if (USE_DEVIL)
     list(APPEND THIS_LIB_LINK_LIBS ${IL_LIBRARY} ${ILUT_LIBRARY})
 else ()
--- a/FreeOrion/GG/GG/PluginInterface.h
+++ b/FreeOrion/GG/GG/PluginInterface.h
@@ -31,13 +31,9 @@
 #define _GG_PluginInterface_h_
 
 #include <GG/GUI.h>
 
-#if defined(__APPLE__) && defined(__MACH__)
-# include "../libltdl/ltdl.h"
-#else
-# include <GG/ltdl.h>
-#endif
+#include <ltdl.h>
 
 #include <string>
 
 
